[CI] CoordinatorTests testRepeatableTests failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/sdiktzidpqmns/tests/:server:test/org.elasticsearch.cluster.coordination.CoordinatorTests/testRepeatableTests

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.CoordinatorTests.testRepeatableTests" -Dtests.seed=BE654CAE797BFA17 -Dtests.locale=bg -Dtests.timezone=Europe/Ulyanovsk -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.CoordinatorTests&tests.test=testRepeatableTests

**Failure excerpt:**
```
java.lang.AssertionError: java.lang.IllegalStateException: recycler entry already released...

  at __randomizedtesting.SeedInfo.seed([BE654CAE797BFA17:5CE72E7F07107FE7]:0)
  at org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:67)
  at org.elasticsearch.common.bytes.ReleasableBytesReference$RefCountedReleasable.closeInternal(ReleasableBytesReference.java:226)
  at org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:53)
  at org.elasticsearch.common.bytes.ReleasableBytesReference.decRef(ReleasableBytesReference.java:68)
  at org.elasticsearch.transport.BytesTransportRequest.decRef(BytesTransportRequest.java:75)
  at org.elasticsearch.test.disruption.DisruptableMockTransport$2.run(DisruptableMockTransport.java:157)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue$2.run(DeterministicTaskQueue.java:516)
  at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster$ClusterNode$2.run(AbstractCoordinatorTestCase.java:1338)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runTask(DeterministicTaskQueue.java:144)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runRandomTask(DeterministicTaskQueue.java:138)
  at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:499)
  at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:354)
  at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase.lambda$testRepeatableTests$0(AbstractCoordinatorTestCase.java:185)
  at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:187)
  at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:211)
  at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase.testRepeatableTests(AbstractCoordinatorTestCase.java:194)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.IllegalStateException: recycler entry already released...

    at org.elasticsearch.common.recycler.NoneRecycler$NV.close(NoneRecycler.java:43)
    at org.elasticsearch.transport.BytesRefRecycler$1.close(BytesRefRecycler.java:42)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$ClearableRecycler$1.close(AbstractCoordinatorTestCase.java:1863)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:75)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:117)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:100)
    at org.elasticsearch.core.Releasables.close(Releasables.java:25)
    at org.elasticsearch.core.Releasables.close(Releasables.java:35)
    at org.elasticsearch.common.io.stream.RecyclerBytesStreamOutput.close(RecyclerBytesStreamOutput.java:174)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:75)
    at org.elasticsearch.core.Releasables.close(Releasables.java:41)
    at org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:65)
    at org.elasticsearch.common.bytes.ReleasableBytesReference$RefCountedReleasable.closeInternal(ReleasableBytesReference.java:226)
    at org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:53)
    at org.elasticsearch.common.bytes.ReleasableBytesReference.decRef(ReleasableBytesReference.java:68)
    at org.elasticsearch.transport.BytesTransportRequest.decRef(BytesTransportRequest.java:75)
    at org.elasticsearch.test.disruption.DisruptableMockTransport$2.run(DisruptableMockTransport.java:157)
    at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue$2.run(DeterministicTaskQueue.java:516)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster$ClusterNode$2.run(AbstractCoordinatorTestCase.java:1338)
    at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runTask(DeterministicTaskQueue.java:144)
    at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runRandomTask(DeterministicTaskQueue.java:138)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:499)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:354)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase.lambda$testRepeatableTests$0(AbstractCoordinatorTestCase.java:185)
    at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:187)
    at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:211)
    at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase.testRepeatableTests(AbstractCoordinatorTestCase.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java
@@ -1860,8 +1860,9 @@ public class AbstractCoordinatorTestCase extends ESTestCase {
 
                 @Override
                 public void close() {
-                    innerRef.close();
-                    trackedRefs.remove(this);
+                    if (trackedRefs.remove(this)) {
+                        innerRef.close();
+                    }
                 }
             };
             trackedRefs.add(trackedRef);

